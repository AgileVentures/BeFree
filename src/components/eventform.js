/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from "react"
import { gql, useMutation } from "@apollo/client"
import DateTimePicker from "react-datetime-picker"
import ProjectSelect from "./projectsselect"

export const CREATE_EVENT = gql`
  mutation CreateEvent($event: CreateEventInput!) {
    createEvent(input: $event) {
      id
      slug
    }
  }
`

const EventForm = props => {
  const [startDate, onStartDatetimeChange] = useState(new Date())
  const [endDate, onEndDatetimeChange] = useState(new Date())
  const [formState, setFormState] = useState({
    name: "",
    category: "PairProgramming",
    projectId: 0,
    description: "",
    timeZone: "UTC",
    duration: 0,
    repeats: "never",
    repeatEnds: "never",
    repeatsEveryNWeeks: 1,
    monday: false,
    tuesday: false,
    wednesday: false,
    thursday: false,
    friday: false,
    saturday: false,
    sunday: false,
  })

  function getMask() {
    let mask = 0
    if (formState.monday) {
      mask += 1
    }
    if (formState.tuesday) {
      mask += 2
    }
    if (formState.wednesday) {
      mask += 4
    }
    if (formState.thursday) {
      mask += 8
    }
    if (formState.friday) {
      mask += 16
    }
    if (formState.saturday) {
      mask += 32
    }
    if (formState.sunday) {
      mask += 64
    }
    return mask
  }

  const [createEvent] = useMutation(CREATE_EVENT, {
    variables: {
      event: {
        name: formState.name,
        description: formState.description,
        category: formState.category,
        timeZone: formState.timeZone,
        repeats: formState.repeats,
        startDatetime: startDate,
        duration: parseInt(formState.duration, 10),
        repeatEnds: false,
        repeatsEveryNWeeks: 1,
        repeatsWeeklyEachDaysOfTheWeekMask: getMask(),
        // endDateTime: endDate,
        clientMutationId: "web",
      },
    },
    onCompleted(data) {
      if (typeof window !== `undefined`)
        window.location.replace(`/events/${data.createEvent.slug}`)
    },
  })

  return (
    <div>
      <form
        onSubmit={e => {
          e.preventDefault()
          createEvent()
        }}
      >
        <div className="grid md:grid-cols-6 gap-4">
          <div className="col-span-4">
            <label className="text-xl font-semibold" htmlFor="name">
              Name
            </label>
            <input
              className="mt-1 block w-full rounded-md border-gray-300 shadow-sm"
              type="text"
              id="name"
              value={formState.name}
              onChange={e =>
                setFormState({
                  ...formState,
                  name: e.target.value,
                })
              }
              required
            />
            <label className="text-xl font-semibold" htmlFor="description">
              Description
            </label>
            <textarea
              className="mt-1 block w-full rounded-md border-gray-300 shadow-sm"
              type="text"
              id="description"
              value={formState.description}
              onChange={e =>
                setFormState({
                  ...formState,
                  description: e.target.value,
                })
              }
            />
            <div className="text-xl font-semibold field">Start DateTime</div>
            <DateTimePicker
              onChange={onStartDatetimeChange}
              value={startDate}
            />
            <br />
            <label className="text-xl font-semibold" htmlFor="duration">
              Duration in minutes
            </label>
            <input
              className="mt-1 block rounded-md border-gray-300 shadow-sm"
              id="duration"
              type="number"
              value={formState.duration}
              onChange={e =>
                setFormState({
                  ...formState,
                  duration: e.target.value,
                })
              }
              required
            />
            <label className="text-xl font-semibold" htmlFor="repeats">
              Repeats
            </label>
            <select
              id="repeats"
              className="mt-1 block rounded-md border-gray-300 shadow-sm"
              value={formState.repeats}
              onChange={e =>
                setFormState({
                  ...formState,
                  repeats: e.target.value,
                })
              }
              required
            >
              <option value="never">never</option>
              <option value="weekly">weekly</option>
              <option value="biweekly">biweekly</option>
            </select>
            <label className="text-xl font-semibold mr-1" htmlFor="monday">
              Monday
            </label>
            <input
              id="monday"
              type="checkbox"
              value={formState.monday}
              onChange={e =>
                setFormState({
                  ...formState,
                  monday: e.target.checked,
                })
              }
            />
            <br />
            <label className="text-xl font-semibold mr-1" htmlFor="tuesday">
              Tuesday
            </label>
            <input
              id="tuesday"
              type="checkbox"
              value={formState.tuesday}
              onChange={e =>
                setFormState({
                  ...formState,
                  tuesday: e.target.checked,
                })
              }
            />
            <br />
            <label className="text-xl font-semibold mr-1" htmlFor="wednesday">
              Wednesday
            </label>
            <input
              id="wednesday"
              name="wednesday"
              type="checkbox"
              value={formState.wednesday}
              onChange={e =>
                setFormState({
                  ...formState,
                  wednesday: e.target.checked,
                })
              }
            />
            <br />
            <label className="text-xl font-semibold mr-1" htmlFor="thursday">
              Thursday
            </label>
            <input
              id="thursday"
              name="thursday"
              type="checkbox"
              value={formState.thursday}
              onChange={e =>
                setFormState({
                  ...formState,
                  thursday: e.target.checked,
                })
              }
            />
            <br />
            <label className="text-xl font-semibold mr-1" htmlFor="friday">
              Friday
            </label>
            <input
              id="friday"
              name="friday"
              type="checkbox"
              value={formState.friday}
              onChange={e =>
                setFormState({
                  ...formState,
                  friday: e.target.checked,
                })
              }
            />
            <br />
            <label className="text-xl font-semibold mr-1" htmlFor="saturday">
              Saturday
            </label>
            <input
              id="saturday"
              name="saturday"
              type="checkbox"
              value={formState.saturday}
              onChange={e =>
                setFormState({
                  ...formState,
                  saturday: e.target.checked,
                })
              }
            />
            <br />
            <label className="text-xl font-semibold mr-1" htmlFor="sunday">
              Sunday
            </label>
            <input
              id="sunday"
              name="sunday"
              type="checkbox"
              value={formState.sunday}
              onChange={e =>
                setFormState({
                  ...formState,
                  sunday: e.target.checked,
                })
              }
            />
            <br />
            <label className="text-xl font-semibold" htmlFor="repeatends">
              Repeat Ends
            </label>
            <select
              id="repeatends"
              className="mt-1 block rounded-md border-gray-300 shadow-sm"
              value={formState.repeatEnds}
              onChange={e =>
                setFormState({
                  ...formState,
                  repeatEnds: e.target.value,
                })
              }
              required
            >
              <option value="on">on</option>
              <option value="never">never</option>
            </select>
            <div className="text-xl font-semibold field">End DateTime</div>
            <DateTimePicker onChange={onEndDatetimeChange} value={endDate} />
          </div>
          <div className="col-span-2">
            <label className="text-xl font-semibold" htmlFor="category">
              Category
            </label>
            <select
              id="category"
              className="mt-1 block w-full rounded-md border-gray-300 shadow-sm"
              value={formState.category}
              onChange={e =>
                setFormState({
                  ...formState,
                  category: e.target.value,
                })
              }
              required
            >
              <option value="PairProgramming">PairProgramming</option>
              <option value="Scrum">Scrum</option>
              <option value="ClientMeeting">ClientMeeting</option>
            </select>
            <div className="text-xl font-semibold field">Project</div>
            <br />
            <ProjectSelect
              value={formState.projectID}
              onChange={e =>
                setFormState({
                  ...formState,
                  projectID: e.target.value,
                })
              }
            />
          </div>
        </div>
        <button
          type="button"
          className="bg-header hover:bg-grey-dark font-bold py-2 px-4 m-2 rounded-r rounded-lg shadow"
        >
          Cancel
        </button>
        <button
          type="submit"
          className="bg-header hover:bg-grey-dark font-bold py-2 px-4 m-2 rounded-r rounded-lg shadow"
        >
          Save
        </button>
      </form>
    </div>
  )
}
export default EventForm
