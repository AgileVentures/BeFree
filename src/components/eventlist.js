import React from "react"
import { StaticQuery, graphql } from "gatsby"

export default function EventList() {
  return (
    <StaticQuery
      query={graphql`
        {
          av {
            events {
              id
              name
              repeats
              startDatetime
              timeZone
              duration
              category
            }
          }
        }
      `}
      render={data => (
        <div>
          {data.av.events.map(e => (
            <h2>{e.name}</h2>
          ))}
        </div>
      )}
    />
  )
}
