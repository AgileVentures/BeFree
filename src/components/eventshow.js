import React from "react"
import { DateTime } from "luxon"
import { useQuery, gql } from "@apollo/client"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import "../utils/font-awesome"

const EVENT = gql`
  query showEvent($slug: String!) {
    event(slug: $slug) {
      id
      name
      repeats
      startDatetime
      timeZone
      duration
      category
      description
    }
  }
`

export const PureEventShow = ({ data }) => (
  <div>
    <h2 className="text-2xl font-bold">{data.event.name}</h2>
    <div>{data.event.description}</div>
    <div>Event type: {data.event.category}</div>
    <FontAwesomeIcon icon="calendar" />
    <span className="month text-gray-700">
      {" "}
      {DateTime.fromISO(data.event.startDatetime).toLocaleString(
        DateTime.DATE_MED_WITH_WEEKDAY
      )}
    </span>
    <br />
    <FontAwesomeIcon icon="clock" />
    <span className="text-gray-700">
      {" "}
      {DateTime.fromISO(data.event.startDatetime).toLocaleString(
        DateTime.TIME_SIMPLE
      )}
    </span>
  </div>
)

const EventShow = ({ slug }) => {
  const { loading, error, data } = useQuery(EVENT, {
    variables: { slug },
  })

  if (loading) return <p>Loading...</p>
  if (error) return <p>Error :(</p>

  return <PureEventShow data={data} />
}

export default EventShow
