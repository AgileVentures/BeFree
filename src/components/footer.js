import React from "react"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { faFacebookF, faTwitter } from "@fortawesome/free-brands-svg-icons"

const Footer = () => {
  return (
    <footer className="flex bg-footer">
      <div>
        <h3>Social</h3>
        <ul>
          <li>
            <a
              className="facebook"
              href="https://www.facebook.com/agileventures"
              target="_blank"
              rel="noreferrer"
              style={{ "white-space": `nowrap` }}
            >
              <FontAwesomeIcon icon={faFacebookF} />
              &nbsp;Facebook
            </a>
          </li>
          <li>
            <a
              className="twitter"
              href="https://twitter.com/AgileVentures"
              target="_blank"
              rel="noreferrer"
              style={{ "white-space": `nowrap` }}
            >
              <FontAwesomeIcon icon={faTwitter} />
              Twitter
            </a>
          </li>
        </ul>
      </div>
      <div>
        <h3>Our Sponsors</h3>
        <ul>
          <li>
            <a
              href="https://standuply.com/?utm_source=links&amp;utm_medium=agileventures&amp;utm_campaign=partnership"
              target="_blank"
              rel="noreferrer"
            >
              Standup Bot
            </a>
          </li>
          <li>
            <a
              href="http://craftacademy.se/english"
              target="_blank"
              rel="noreferrer"
            >
              Craft Academy
            </a>
          </li>
          <li>
            <a href="http://mooqita.org/" target="_blank" rel="noreferrer">
              Mooqita
            </a>
          </li>
        </ul>
      </div>
      <div>
        <h3>Contact Us</h3>
        <p>
          Send a traditional email to
          <a href="mailto:info@agileventures.org">info@agileventures.org</a>.
        </p>
        <p>
          We are a Charitable Incorporated Organisation (CIO) in the UK. Ref
          #1170963
        </p>
      </div>
    </footer>
  )
}

export default Footer
