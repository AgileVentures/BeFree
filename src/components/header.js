import React from "react"

import logo from "../../static/logo.png"

const Header = () => {
  return (
    <header className="flex h-16 bg-header">
      <a href="/" className="shadow-none flex-1 text-center px-4 py-2 m-2">
        <img id="logo" data-testid="logo" src={logo} alt="Agile Ventures" />
      </a>
    </header>
  )
}

export default Header
