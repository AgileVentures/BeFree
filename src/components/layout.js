import React from "react"
import { ApolloProvider, ApolloClient, InMemoryCache } from "@apollo/client"
import { rhythm } from "../utils/typography"
import Header from "./header"

const client = new ApolloClient({
  uri: process.env.REACT_APP_AGILEVENTURES_API,
  cache: new InMemoryCache(),
})

const Layout = ({ children }) => (
  <ApolloProvider client={client}>
    <div>
      <Header />
      <div
        style={{
          marginLeft: `auto`,
          marginRight: `auto`,
          maxWidth: rhythm(24),
          padding: `${rhythm(1.5)} ${rhythm(3 / 4)}`,
        }}
      >
        <main>{children}</main>
        <footer>
          Â© {new Date().getFullYear()}, Built with
          {` `}
          <a href="https://www.gatsbyjs.org">Gatsby</a>
        </footer>
      </div>
    </div>
  </ApolloProvider>
)

export default Layout
