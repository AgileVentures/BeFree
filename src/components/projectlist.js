import React from "react"
import { useQuery, gql } from "@apollo/client"

const PROJECTS_LIST = gql`
  query listProjects {
    projects {
      title
      id
      slug
    }
  }
`

export const PureProjectList = ({ data }) => (
  <div>
    {data.projects.map(p => (
      <div
        className="bg-white rounded shadow border p-6 mb-4 mt-0"
        key={`project-${p.id}`}
      >
        <h2 className="text-2xl font-bold">
          <a href={`/projects/${p.slug}`}>{p.title}</a>
        </h2>
        <div>{p.description}</div>
      </div>
    ))}
  </div>
)

export default function ProjectList() {
  const { loading, error, data } = useQuery(PROJECTS_LIST)

  if (loading) return <p>Loading...</p>
  if (error) return <p>Error :(</p>

  return <PureProjectList data={data} />
}
