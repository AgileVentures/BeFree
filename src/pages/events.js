import React from "react"
import DynamicEventList from "../components/eventlist"
import Layout from "../components/layout"
import SEO from "../components/seo"

const EventIndex = ({ data, location }) => {
  return (
    <Layout location={location} title="Events">
      <SEO title="All events" />
      <header>
        <h1 className="text-2xl font-bold">AgileVentures Events</h1>
        <div className="mb-4">
          We host several events a day using different services. Feel free to
          join in to get involved or if you are curious about pair programming
          or Agile Ventures.
          <span className="font-bold">
            {" "}
            Each event will have a link to the hangout below.
          </span>{" "}
          You can also add events to your calendar.
        </div>
        <DynamicEventList />
      </header>
    </Layout>
  )
}

export default EventIndex
