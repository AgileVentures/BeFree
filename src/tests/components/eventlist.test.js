import React from "react"
import renderer from "react-test-renderer"
import { PureEventList as EventList } from "../../components/eventlist"
import events from "../fixtures/events"

describe("EventList", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<EventList data={events} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
