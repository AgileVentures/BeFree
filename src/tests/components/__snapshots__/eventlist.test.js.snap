// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList renders correctly 1`] = `
<div>
  <div
    className="bg-white rounded shadow border p-6 mb-4 mt-0"
  >
    <h2
      className="text-2xl font-bold"
    >
      <a
        href="/events/premium-event"
      >
        Premium event
      </a>
    </h2>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-calendar fa-w-14 "
      data-icon="calendar"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="month text-gray-700"
    >
       
      Thu, Nov 26, 2020
    </span>
    <br />
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-clock fa-w-16 "
      data-icon="clock"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="text-gray-700"
    >
       
      11:45 AM
      -
      1:05 AM
    </span>
    <div />
  </div>
  <div
    className="bg-white rounded shadow border p-6 mb-4 mt-0"
  >
    <h2
      className="text-2xl font-bold"
    >
      <a
        href="/events/morning-event"
      >
        Morning event
      </a>
    </h2>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-calendar fa-w-14 "
      data-icon="calendar"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="month text-gray-700"
    >
       
      Thu, Nov 26, 2020
    </span>
    <br />
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-clock fa-w-16 "
      data-icon="clock"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="text-gray-700"
    >
       
      12:15 PM
      -
      1:35 AM
    </span>
    <div />
  </div>
  <div
    className="bg-white rounded shadow border p-6 mb-4 mt-0"
  >
    <h2
      className="text-2xl font-bold"
    >
      <a
        href="/events/evening-event"
      >
        Evening event
      </a>
    </h2>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-calendar fa-w-14 "
      data-icon="calendar"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="month text-gray-700"
    >
       
      Thu, Nov 26, 2020
    </span>
    <br />
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-clock fa-w-16 "
      data-icon="clock"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="text-gray-700"
    >
       
      5:45 PM
      -
      8:25 PM
    </span>
    <div />
  </div>
  <div
    className="bg-white rounded shadow border p-6 mb-4 mt-0"
  >
    <h2
      className="text-2xl font-bold"
    >
      <a
        href="/events/premium-event"
      >
        Premium event
      </a>
    </h2>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-calendar fa-w-14 "
      data-icon="calendar"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="month text-gray-700"
    >
       
      Fri, Nov 27, 2020
    </span>
    <br />
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-clock fa-w-16 "
      data-icon="clock"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="text-gray-700"
    >
       
      11:45 AM
      -
      1:05 AM
    </span>
    <div />
  </div>
  <div
    className="bg-white rounded shadow border p-6 mb-4 mt-0"
  >
    <h2
      className="text-2xl font-bold"
    >
      <a
        href="/events/morning-event"
      >
        Morning event
      </a>
    </h2>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-calendar fa-w-14 "
      data-icon="calendar"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="month text-gray-700"
    >
       
      Fri, Nov 27, 2020
    </span>
    <br />
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-clock fa-w-16 "
      data-icon="clock"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="text-gray-700"
    >
       
      12:15 PM
      -
      1:35 AM
    </span>
    <div />
  </div>
  <div
    className="bg-white rounded shadow border p-6 mb-4 mt-0"
  >
    <h2
      className="text-2xl font-bold"
    >
      <a
        href="/events/evening-event"
      >
        Evening event
      </a>
    </h2>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-calendar fa-w-14 "
      data-icon="calendar"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="month text-gray-700"
    >
       
      Fri, Nov 27, 2020
    </span>
    <br />
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-clock fa-w-16 "
      data-icon="clock"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="text-gray-700"
    >
       
      5:45 PM
      -
      8:25 PM
    </span>
    <div />
  </div>
</div>
`;
